src/Service.Tests/GraphQLBuilder/NestedMutationBuilderTests.cs

```diff
@@ -0,0 +1,408 @@
+        }
+
+        /// <summary>
+        /// Test to validate that that we create a source -> target linking object definition for every pair of (source, target) entities which
+        /// are related via an M:N relationship.
+        /// </summary>
+        /// <param name="sourceEntityName">Name of the source entity for which the configuration is provided in the config.</param>
+        /// <param name="targetEntityName">Name of the target entity which is related to the source entity via a relationship defined in the 'relationships'
+        /// section in the configuration of the source entity.</param>
+        [DataTestMethod]
+        [DataRow("Book", "Author", DisplayName = "Validate presence of source->target linking object for Book->Author M:N relationship")]
+        [DataRow("Author", "Book", DisplayName = "Validate presence of source->target linking object for Author->Book M:N relationship")]
+        [DataRow("BookNF", "AuthorNF", DisplayName = "Validate presence of source->target linking object for BookNF->AuthorNF M:N relationship")]
+        [DataRow("AuthorNF", "BookNF", DisplayName = "Validate presence of source->target linking object for AuthorNF->BookNF M:N relationship")]
+        public void ValidatePresenceOfSourceTargetLinkingObjectDefinitionsInObjectsNodeForMNRelationships(string sourceEntityName, string targetEntityName)
+        {
+            string sourceTargetLinkingNodeName = GenerateLinkingNodeName(
+                        GetDefinedSingularName(sourceEntityName, _runtimeConfig.Entities[sourceEntityName]),
+                        GetDefinedSingularName(targetEntityName, _runtimeConfig.Entities[targetEntityName]));
+            ObjectTypeDefinitionNode sourceTargetLinkingObjectTypeDefinitionNode = GetObjectTypeDefinitionNode(sourceTargetLinkingNodeName);
+
+            // Validate that we have indeed inferred the object type definition for the source->target linking object.
+            Assert.IsNotNull(sourceTargetLinkingObjectTypeDefinitionNode);
+        }
+
+        /// <summary>
+        /// Test to validate that we add a Foriegn key directive to the list of directives for every column in an entity/table,
+        /// which holds a foreign key reference to some other entity in the config.
+        /// </summary>
+        /// <param name="referencingEntityName">Name of the referencing entity.</param>
+        /// <param name="referencingColumns">List of referencing columns.</param>
+        [DataTestMethod]
+        [DataRow("Book", new string[] { "publisher_id" },
+            DisplayName = "Validate referencing field directive for referencing columns in Book entity for Book->Publisher relationship.")]
+        [DataRow("Review", new string[] { "book_id" },
+            DisplayName = "Validate referencing field directive for referencing columns in Review entity for Review->Book relationship.")]
+        [DataRow("stocks_price", new string[] { "categoryid", "pieceid" },
```

Comment:
Do we not add any referencing directive for the foreign key fields in the SourceTargetLinkingNode?

Reply:
You are talking about M:N relationships where the sourceTargetLinkingNode contains fields from target node and linking node.

The fields from target node are copied as is, and if any of the target node fields contain redferencing field directive, they will retain it. 

The fields from linking node never contain the directive - since it is not required to do an insertion in the linking entity directly. More here; https://github.com/Azure/data-api-builder/pull/1902#discussion_r1503754427